### PR TITLE
Organization classifications

### LINKED ISSUE #436: Organization classifications: Geographic scope and high-level sector
# Background

I'm breaking this out from #413 so as not to hold up preparing a PR for that issue.

This issue relates to the following CoST IDS elements proposed in the CoST IDS/OC4IDS review:

<details>

<summary> Accredited Entity Type (preparation)   </summary>

## Accredited Entity Type (preparation) 

**Module:** Climate finance
**Indicator:** Accessibility

### Disclosure format

> Select from the lists: 
> * Private 
> * Public 
> * non-governmental 
> * sub-national 
> * national 
> * regional 
> * international 

### OC4IDS mapping

> TBC

</details>

I've copied the relevant comments from https://github.com/open-contracting/ocds-extensions/issues/207 below.

@duncandewhurst:

> CoST suggested the following classification for financing party type, which we'll need to reconcile with the existing codelist:
> 
> > * Private
> > * Public
> > * non-governmental
> > * sub-national
> > * national
> > * regional
> > * international
> 
> We might consider splitting into separate fields or classifications since the above list mixes two concepts: high-level sector (private, public, non-governmental) and geographic scope (sub-national, national, regional, international).
> 
> @EvelynDinora provided this helpful explanation of the use cases for the classification:
> 
> > the interest of the SG is to see which and what type of entity is facilitating the investments in climate finance projects in the country. The experience tells that major entities such multilateral banks are obtaining funds against national entities. The issue is that major entities are extremely burocratic and the transactional costs are higher than the national entities who can speed up the investments at lower costs (again aggregated data is the main interest but this collected project per project).

@jpmckinney:

> Agree on splitting out geographic scope, etc. These would be classifications on the organization entry in the `parties` array?

@duncandewhurst:

> Yes, I think that makes sense.
> 
> In [#207 (comment)](https://github.com/open-contracting/ocds-extensions/issues/207#issuecomment-1765456721), I was concerned that the details of the actual organisation might not be known/disclosable, but the CoST IDS/OC4IDS review proposes an element for that.

# Discussion

@jpmckinney a couple of questions:

1. Was your suggestion to model this using the [OCDS organization classification extension](https://extensions.open-contracting.org/en/extensions/organizationClassification/1.1/)?
2. If so, are you happy with the `scheme` values below? (If you have a suggestion for a better name for the private/public/non-governmental classification that would be very welcome)

```json
{
  "parties": [
    {
      "details": {
        "classifications": [
          {
            "id": "private",
            "scheme": "COST_IDS_SECTOR"
          },
          {
            "id": "national",
            "scheme": "COST_IDS_GEOGRAPHIC_SCOPE"
          }
        ]
      }
    }
  ]
}
```

cc @evelyndinora @mgraca-prado

### PR DESCRIPTION
**Pull request by duncandewhurst**
**Related issues**

* Closes #436 

**Description**

**Merge checklist**

- [x] [Log your changes](https://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/contributing.html#logging-changes)

If there are changes to `project-schema.json` or `project-package-schema.json`:

- [x] Update the examples:
  - [x] `docs/examples/example.json`
  - [x] `docs/examples/blank.json`
- [x] Run `./manage.py pre-commit` to update `docs/_static/i8n.csv`

If you added a new definition to the schema, run `./manage.py pre-commit`.

If you added a new codelist:

  - [ ] Add an entry to `docs/reference/codelists.md`

### CODE REVIEW

**Review by odscjen — APPROVED** [review]
LGTM

File: schema/project-level/codelists/classificationScheme.csv [hunk 1]
```diff
@@ -1,3 +1,5 @@
 Code,Title,Description,Source,Category
 sdg,Sustainable Development Goals,"The Sustainable Development Goals are a collection of 17 interlinked goals set by the United Nations General Assembly and designed to be a ""blueprint to achieve a better and more sustainable future for all"".",https://sdgs.un.org/goals,purpose
 sdgTarget,Sustainable Development Goal Targets,The Sustainable Development Goal Targets are specific targets for each Sustainable Development Goal.,https://sdgs.un.org/goals,purpose
+costIdsLegalType,CoST IDS Legal Type,The disclosure format for the accredited entity type CoST IDS element provides a classification of an organization's legal type.,https://standard.open-contracting.org/infrastructure/latest/en/cost/ids/sustainability/#climate-finance-accredited-entity-type,organization
+costIdsAdministrativeLevel,CoST IDS Administrative Level,The disclosure format for the accredited entity type CoST IDS element provides a classification of an organization's administrative level.,https://standard.open-contracting.org/infrastructure/latest/en/cost/ids/sustainability/#climate-finance-accredited-entity-type,organization
```
Comment (jpmckinney) [inline schema/project-level/codelists/classificationScheme.csv]: I have trouble understanding "The disclosure format for the accredited entity type CoST IDS element provides"

Is there some shorter or alternative version?

File: schema/project-level/project-schema.json [hunk 2]
```diff
@@ -1303,6 +1304,27 @@
           "uniqueItems": true,
           "minItems": 1
         },
+        "details": {
```
Comment (jpmckinney) [inline schema/project-level/project-schema.json]: I find `details` to be a bit of a schema anti-pattern, so I would prefer not to mirror OCDS here.

If kept, its current `description` is specific to OCDS.

File: schema/project-level/project-schema.json [hunk 3]
```diff
@@ -1303,6 +1304,27 @@
...
+          "type": [
+            "object"
+          ],
+          "description": "Additional classification information about parties can be provided using partyDetail extensions that define particular fields and classification schemes.",
+          "title": "Details",
+          "properties": {
+            "classifications": {
+              "title": "Organization classifications",
```
Comment (jpmckinney) [inline schema/project-level/project-schema.json]:
```suggestion
              "title": "Classifications",
```

File: schema/project-level/project-schema.json [hunk 4]
```diff
@@ -1303,6 +1304,27 @@
...
+              "description": "The classifications that categorize the organization",
```
Comment (jpmckinney) [inline schema/project-level/project-schema.json]:
```suggestion
              "description": "The classifications that categorize the organization.",
```

File: schema/project-level/project-schema.json [hunk 5]
```diff
@@ -1303,6 +1304,27 @@
...
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Classification"
+              },
+              "wholeListMerge": true,
```
Comment (jpmckinney) [inline schema/project-level/project-schema.json]:
```suggestion
```

File: schema/project-level/codelists/classificationScheme.csv (hunk 1 above)
Comment (duncandewhurst) [reply schema/project-level/codelists/classificationScheme.csv]: Given that the source column provides a direct link to the CoST IDS element in which the classification is specified, I think we could just have:

> The CoST IDS classification of an organization's legal type.

and

> The CoST IDS classification of an organization's administrative level.

Sound good?

File: schema/project-level/project-schema.json (hunk 2 above)
Comment (duncandewhurst) [reply schema/project-level/project-schema.json]: How about we just have `classifications` directly on `Organization` instead? I don't know why it's nested under `details` in OCDS.

File: schema/project-level/codelists/classificationScheme.csv (hunk 1 above)
Comment (jpmckinney) [reply schema/project-level/codelists/classificationScheme.csv]: Yes, much better!

File: schema/project-level/project-schema.json (hunk 2 above)
Comment (jpmckinney) [reply schema/project-level/project-schema.json]: Yes, let's put it direct. The original issue is https://github.com/open-contracting/standard/issues/369 but there's not much motivation for having a `details` object.

File: schema/project-level/codelists/classificationScheme.csv (hunk 1 above)
Comment (duncandewhurst) [reply schema/project-level/codelists/classificationScheme.csv]: Done in 80f90799bcaba37f5e8efe3d9b26a0e848ed2329

File: schema/project-level/project-schema.json (hunk 2 above)
Comment (duncandewhurst) [reply schema/project-level/project-schema.json]: Done in https://github.com/open-contracting/infrastructure/pull/445/commits/59f5b6b7c85d88e58bddb063fa8138e347817aac

File: docs/reference/changelog.md [hunk 6]
```diff
@@ -15,7 +15,7 @@
 * [#422](https://github.com/open-contracting/infrastructure/pull/422), [#425](https://github.com/open-contracting/infrastructure/pull/425) - add CoST IDS sustainability modules to mapping documentation.
 
 ### Schema
-
+https://github.com/open-contracting/infrastructure/pull/445/commits/59f5b6b7c85d88e58bddb063fa8138e347817aac
```
Comment (jpmckinney) [inline docs/reference/changelog.md]: Bad merge? (Restore newline)

File: schema/project-level/project-schema.json [hunk 7]
```diff
@@ -1388,6 +1389,17 @@
           "uniqueItems": true,
           "minItems": 1
         },
+        "classifications": {
+          "title": "Organization classifications",
+          "description": "The classifications that categorize the organization",
```
Comment (jpmckinney) [inline schema/project-level/project-schema.json]:
```suggestion
          "description": "The classifications that categorize the organization.",
```

**Review by jpmckinney — APPROVED** [review]
Ok to merge if ~~two~~ three small comments resolved.

File: schema/project-level/project-schema.json [hunk 8]
```diff
@@ -1388,6 +1389,17 @@
...
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Classification"
+          },
+          "wholeListMerge": true,
```
Comment (jpmckinney) [inline schema/project-level/project-schema.json]:
```suggestion
```

File: docs/reference/changelog.md (hunk 6 above)
Comment (duncandewhurst) [reply docs/reference/changelog.md]: An accidental paste, by the look of it. Fixed in https://github.com/open-contracting/infrastructure/pull/445/commits/d226f235aec4d1a226b2567885391d9553e35d3c. Thanks for flagging

**Pull request merged**